From: Kurt Roeckx <kurt@roeckx.be>
Date: Fri, 4 Dec 2015 16:20:43 +0100
Subject: [PATCH] Disable the freelist

We don't define OPENSSL_NO_BUF_FREELISTS globally sinc it changes structures and
would break the ABI.  Instead we just do it in the .c files that try to do
something with it.


Imported from openssl_1.0.2e-1.debian.tar.xz

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 ssl/s3_both.c | 1 +
 ssl/ssl_lib.c | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/ssl/s3_both.c b/ssl/s3_both.c
index 09d0661e81f6..3429899872b8 100644
--- a/ssl/s3_both.c
+++ b/ssl/s3_both.c
@@ -573,6 +573,7 @@ int ssl_verify_alarm_type(long type)
     return (al);
 }
 
+#define OPENSSL_NO_BUF_FREELISTS
 #ifndef OPENSSL_NO_BUF_FREELISTS
 /*-
  * On some platforms, malloc() performance is bad enough that you can't just
diff --git a/ssl/ssl_lib.c b/ssl/ssl_lib.c
index f2071db93702..c605784474a5 100644
--- a/ssl/ssl_lib.c
+++ b/ssl/ssl_lib.c
@@ -162,6 +162,8 @@
 
 const char *SSL_version_str = OPENSSL_VERSION_TEXT;
 
+#define OPENSSL_NO_BUF_FREELISTS
+
 SSL3_ENC_METHOD ssl3_undef_enc_method = {
     /*
      * evil casts, but these functions are only called if there's a library
